src/core/utils/compatibility.js

```diff
@@ -113,7 +113,9 @@ export function isCompatibleWithUserAgent({
     if (
       addon.type === ADDON_TYPE_EXTENSION &&
       !findInstallURL({
-        platformFiles: addon.platformFiles, userAgentInfo,
+        appendSource: false,
```

Comment:
Ah, okay, I see... seems we do set this as false a fair bit. So I guess ignore my earlier comment about restructuring the `let source` bit 😅